zerver/lib/email_mirror.py

```diff
@@ -186,23 +187,21 @@ def get_message_part_by_type(message: message.Message, content_type: str) -> Opt
     return None
 
 talon_initialized = False
-def extract_body(message: message.Message, include_quotes: bool=False) -> str:
+def extract_body(message: message.Message, include_quotes: bool=False, prefer_text: bool=True) -> str:
     import talon
     global talon_initialized
     if not talon_initialized:
         talon.init()
         talon_initialized = True
 
-    # If the message contains a plaintext version of the body, use
-    # that.
     plaintext_content = get_message_part_by_type(message, "text/plain")
-    if plaintext_content:
+    if prefer_text and plaintext_content:
```

Comment:
I think this won't correctly handle `prefer_text=False` and only plaintext content being available.  We probably need an `if (prefer_text or not html_content) and plaintext_content` sort of thing.

Reply:
@timabbott My idea was to treat ``prefer-html`` and no html body as something that shouldn't be done - but yeah, it's more consistent to just fall back to plaintext. I pushed a commit to add that behavior. It required changing up the ``extract_body`` quite a bit, so I think it's cleaner to have that as the additional commit, rather than squashing.